source/initial_temperature/adiabatic.cc

```diff
@@ -33,17 +33,47 @@ namespace aspect
 {
   namespace InitialTemperature
   {
+
+    template <int dim>
+    Adiabatic<dim>::Adiabatic ()
+      :
+      surface_boundary_id(numbers::invalid_unsigned_int)
+    {}
+
+    template <int dim>
+    void
+    Adiabatic<dim>::initialize ()
+    {
+      // Find the boundary indicator that represents the surface
+      surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(surface_boundary_id);
+
+      // The input ascii table contains one data column (LAB depths(m)) in addition to the coordinate columns.
+      Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set,
```

Comment:
This is not a full review, but I just noticed this when I looked through the file: You only want to do this in case 
```
if (read_from_ascii_file)
```
otherwise ASPECT will try to read in a data file that may not be there.

Reply:
I tried this and it doesn't allow it because of the following error:

```
cannot assign to non-static data member within const member function 'initial_temperature'
 surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
```
Hopefully this error can be fixed and there's a way around this, but if not I don't think it would matter. If someone were using this plugin and didn't want to use an ASCII file, since the default directory is the file in data/initial-temperature/adiabatic they would automatically have this file when they clone ASPECT and it would never not find the file.

Reply:
I should mention that the error I just sent was what I tried to do after I tried doing what you suggested. When I just added the two lines that you highlighted in your review I received this error:

```
error: use of undeclared identifier 'surface_boundary_set'; did you mean 'surface_boundary_id'?
          surface_boundary_set.insert(surface_boundary_id);
```

Reply:
@danieldouglas92 - note the suggested fix above from @jdannberg